### PR TITLE
In App Feedback Prompt Groundwork

### PR DESCRIPTION
**Pull request by salimbraksa**
Part of #22341

## Description

This PR makes changes to the existing `AppRatingsUtility` type. This utility object is responsible for deciding when the App Ratings UI should be shown. The changes include:
1. Remove the existing App Ratings UX from the Notifications screen.
2. Remove the old remote feature flag mechanism this object was using.
3. Raise the required action count for the App Ratings Prompt to display to 20.
4. The following actions increment the action count:
    - ~User visits the Notifications screens and has unread notifications.~ User taps on unread notification ( see this [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/22052/files#r1446656993) ) 
    - User publishes a post.
    - User reads a post.
    - User views stats.

## Test Instructions
Same as https://github.com/wordpress-mobile/WordPress-iOS/pull/22052

## Notes
Remove then `DO NOT MERGE` label when the following PR is approved:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22052

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mokagio** [comment]
I've been going trough PRs to update them via `git merge trunk` to include #22270 and run on the new Apple Silicon CI with Xcode 15.1.

I did not work on this one because it already developed merge conflicts with `trunk` so a merge and work by the author is required regardless.

File: WordPress/Classes/System/WordPressAppDelegate.swift [hunk 1]
```diff
@@ -411,14 +411,8 @@ extension WordPressAppDelegate {
         }
 
         let utility = AppRatingUtility.shared
-        utility.register(section: "notifications", significantEventCount: 5)
-        utility.systemWideSignificantEventCountRequiredForPrompt = 10
+        utility.systemWideSignificantEventCountRequiredForPrompt = 20
```
Comment (salimbraksa) [inline WordPress/Classes/System/WordPressAppDelegate.swift:411]: This means, that at least 20 actions should be performed for the App Ratings Prompt to appear.

The count is increased using this method:

```swift
AppRatingUtility.shared.incrementSignificantEvent()
```

File: WordPress/Classes/Utility/Ratings/AppRatingsUtility.swift [hunk 2]
```diff
@@ -65,54 +71,12 @@ class AppRatingUtility: NSObject {
...
             if shouldSkipRating {
                 checkNewVersionNeedsSkipping()
             }
         }
     }
 
-    /// This checks if we've disabled app review prompts for a feature or at a
-    /// global level
-    ///
-    @objc func checkIfAppReviewPromptsHaveBeenDisabled(success: (() -> Void)?, failure: (() -> Void)?) {
```
Comment (salimbraksa) [inline WordPress/Classes/Utility/Ratings/AppRatingsUtility.swift:78]: This is an old remote feature flag mechanism. It is replaced with the new one.

File: WordPress/Classes/Utility/Ratings/AppRatingsUtility.swift [hunk 3]
```diff
@@ -121,7 +85,7 @@ class AppRatingUtility: NSObject {
     ///
     @objc(registerSection:withSignificantEventCount:)
     func register(section: String, significantEventCount count: Int) {
-        sections[section] = Section(significantEventCount: count, enabled: true)
+        sections[section] = Section(significantEventCount: count)
     }
```
Comment (salimbraksa) [inline WordPress/Classes/Utility/Ratings/AppRatingsUtility.swift:89]: The `enabled` property was never set to `false`. It was always `true`.

**Review by alpavanoglu — COMMENTED** [review]
LGTM. I'll test it again though when we merge the child PR into this one.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/267">24.1</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by alpavanoglu — APPROVED** [review]

**Pull request merged**